paddle/framework/attribute.h

```diff
@@ -39,6 +39,21 @@ AttrType AttrTypeID();
 
 Attribute GetAttrValue(const OpDesc::Attr& attr_desc);
 
+class AttrReader {
```

Comment:
It seems that the purpose of this class is to add a method `Get`, who can find and read an entry in [AttributeMap](https://github.com/PaddlePaddle/Paddle/blob/544458e0115618c82a2181c887b3992a59222844/paddle/framework/attribute.h#L35).

How about we change the definition of AttributeMap from the current one

```c++
typedef std::unordered_map<std::string, Attribute> AttributeMap;
```

into 

```c++
class AttributeMap : public std::unordered_map<std::string, Attribute> {
 public:
  template <typename T> inline const T& Get(...) const { ... }
}
```

so could we define `Get` as a method of type `AttributeMap`; instead of adding a new type `AttrReader`?

Reply:
the problem is if we use a Class that inherit from map, then we can't directly use the [list_initialization](http://en.cppreference.com/w/cpp/language/list_initialization) to init this AttributeMap, try to find a better way to do this in next PR